demo-python/code/azure-search-vector-image-index-creation-python-sample.ipynb

```diff
@@ -0,0 +1,389 @@
+    "This code demonstrates how to create Azure Cognitive Search Index with Azure AI Vision Image Embedding API and Azure Python SDK\n",
+    "## Prerequisites\n",
+    "To run the code, install the following packages. Please use the latest pre-release version `pip install azure-search-documents --pre`. This sample currently uses version `11.4.0b11`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! pip install azure-search-documents==11.4.0b11 --quiet\n",
+    "! pip install python-dotenv ipython --quiet\n",
+    "! pip install azure-storage-blob --quiet"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Import required libraries and environment variables"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import required libraries  \n",
+    "import os  \n",
+    "import json  \n",
+    "import requests\n",
+    "import http.client, urllib.parse\n",
+    "import shutil\n",
+    "from dotenv import load_dotenv  \n",
+    "from azure.core.credentials import AzureKeyCredential  \n",
+    "from azure.search.documents.indexes import SearchIndexClient, SearchIndexerClient  \n",
+    "\n",
+    "from azure.search.documents.indexes.models import (  \n",
+    " \n",
+    "    ExhaustiveKnnParameters,  \n",
+    "    ExhaustiveKnnVectorSearchAlgorithmConfiguration,\n",
+    "    HnswParameters,  \n",
+    "    HnswVectorSearchAlgorithmConfiguration,\n",
+    "    IndexingParameters,\n",
+    "    SimpleField,\n",
+    "    SearchField,  \n",
+    "    SearchFieldDataType,  \n",
+    "    SearchIndex,  \n",
+    "    SearchIndexer,\n",
+    "    SearchIndexerDataSourceConnection, \n",
+    "    SearchIndexerDataContainer,  \n",
+    "    VectorSearch,  \n",
+    "    VectorSearchAlgorithmKind,  \n",
+    "    VectorSearchProfile,  \n",
+    ")\n",
+    "\n",
+    "from azure.storage.blob import BlobServiceClient\n",
+    "  \n",
+    "load_dotenv()  \n",
+    "service_endpoint = os.getenv(\"AZURE_SEARCH_SERVICE_ENDPOINT\")  \n",
+    "index_name = os.getenv(\"AZURE_SEARCH_INDEX_NAME\")\n",
+    "key = os.getenv(\"AZURE_SEARCH_ADMIN_KEY\")  \n",
+    "aiVisionApiKey = os.getenv(\"AZURE_AI_VISION_API_KEY\")  \n",
+    "aiVisionRegion = os.getenv(\"AZURE_AI_VISION_REGION\")\n",
+    "blob_connection_string = os.getenv(\"BLOB_CONNECTION_STRING\")  \n",
+    "container_name = os.getenv(\"BLOB_CONTAINER_NAME\")\n",
+    "credential = AzureKeyCredential(key)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Image Embedding\n",
+    "\n",
+    "Vectorize images with image embedding API "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_image_vector(image_path, key, region):\n",
```

Comment:
Since you are using an indexer - perhaps we need to demonstrate a custom skill that calls the florence embedding API on images in a blob container rather than local uploading. https://learn.microsoft.com/en-us/azure/search/cognitive-search-custom-skill-web-api

Reply:
Good suggestion. As we discussed, we can have two options for the user to choose: local images and images in a blob container.